cli/lsp/language_server.rs

```diff
@@ -3284,9 +3313,10 @@ impl tower_lsp::LanguageServer for LanguageServer {
       {
         return;
       }
-      inner.diagnostics_server.state()
+      inner.diagnostics_state.clone()
     };
-    if !diagnostics_state.read().await.has_no_cache_diagnostic(uri) {
+    // Call this outside of the language server lock, it sync locks internally.
```

Comment:
Oh, I just realized this comment. We can remove it because what it's doing doesn't have the chance to deadlock since it's contained within DiagnosticsState and the lock will be acquired and released really quickly. So it can be called within the language server lock.

Reply:
https://github.com/denoland/deno/pull/20665#discussion_r1335245282